packages/imperative/__tests__/config/cmd/report-env/EnvQuery.unit.test.ts

```diff
@@ -0,0 +1,397 @@
+            expect(itemObj.itemProbMsg).toBe("");
+        });
+
+        it("should report the NVM version", async () => {
+            const itemObj: IGetItemVal = await EnvQuery.getEnvItemVal(ItemId.NVM_VER);
+            if (!itemObj.itemVal.includes("nvm failed to display any output")) {
+                expect(itemObj.itemVal).toMatch(/[0-9]+.[0-9]+.[0-9]+/);
+                expect(itemObj.itemValMsg).toContain("Node Version Manager version =");
+            }
+            expect(itemObj.itemProbMsg).toBe("");
+        });
+
+        it("should report the platform", async () => {
+            const itemObj: IGetItemVal = await EnvQuery.getEnvItemVal(ItemId.PLATFORM);
+            expect(itemObj.itemVal === "win32" || itemObj.itemVal === "linux" || itemObj.itemVal === "darwin").toBeTruthy();
+            expect(itemObj.itemValMsg).toContain("O.S. platform =");
+            expect(itemObj.itemProbMsg).toBe("");
+        });
+
+        it("should report the architecture", async () => {
+            const itemObj: IGetItemVal = await EnvQuery.getEnvItemVal(ItemId.ARCHITECTURE);
+            expect(
+                itemObj.itemVal === "arm"   || itemObj.itemVal === "arm64"  || itemObj.itemVal === "ia32" ||
+                itemObj.itemVal === "mips"  || itemObj.itemVal === "mipsel" || itemObj.itemVal === "ppc" ||
+                itemObj.itemVal === "ppc64" || itemObj.itemVal === "s390"   || itemObj.itemVal === "s390x" ||
+                itemObj.itemVal === "x64"
+            ).toBeTruthy();
+            expect(itemObj.itemValMsg).toContain("O.S. architecture =");
+            expect(itemObj.itemProbMsg).toBe("");
+        });
+
+        it("should report the OS command path", async () => {
+            const itemObj: IGetItemVal = await EnvQuery.getEnvItemVal(ItemId.OS_PATH);
+            expect(itemObj.itemValMsg).toContain("O.S. PATH =");
+            expect(itemObj.itemProbMsg).toBe("");
+        });
+
+        it("should report the ZOWE_CLI_HOME", async () => {
+            // cliHome is a getter property, so mock the property
+            Object.defineProperty(impCfg, "cliHome", {
+                configurable: true,
+                get: jest.fn(() => {
+                    return fakeCliHomeDir;
+                })
+            });
+
+            const itemObj: IGetItemVal = await EnvQuery.getEnvItemVal(ItemId.ZOWE_CLI_HOME);
+            expect(itemObj.itemVal).toContain("undefined");
+            expect(itemObj.itemVal).toContain("Default = " + fakeCliHomeDir);
+            expect(itemObj.itemValMsg).toContain("ZOWE_CLI_HOME =");
+            expect(itemObj.itemProbMsg).toBe("");
+        });
+
+        it("should report an undefined ZOWE_APP_LOG_LEVEL", async () => {
+            const itemObj: IGetItemVal = await EnvQuery.getEnvItemVal(ItemId.ZOWE_APP_LOG_LEVEL);
+            expect(itemObj.itemVal).toBeUndefined();
+            expect(itemObj.itemValMsg).toContain("ZOWE_APP_LOG_LEVEL = undefined");
```

Comment:
I am a bit confused here. Perhaps it is a misunderstanding. This test explicitly confirms the situation when the ZOWE_APP_LOG_LEVEL is not set. Another test confirms what happens when ZOWE_APP_LOG_LEVEL has a valid value, and yet another confirms behavior when ZOWE_APP_LOG_LEVEL has been set to a bogus value.

Reply:
I only brought this up because, when I ran the tests locally, the test failed because I have the log level environment variables set in my .bashrc file. I suppose I can clear it as part of running tests, I was just curious if part of the test should be to clear the log level environment variable, do the tests, and restore it at the end of the tests.

Reply:
Ah, now I understand. I think it is reasonable to ensure that the variable is unset at the beginning of this particular test. I will add that change.